src/librustc/ty/sty.rs

```diff
@@ -57,7 +57,7 @@ pub enum BoundRegion {
     ///
     /// The def-id is needed to distinguish free regions in
     /// the event of shadowing.
-    BrNamed(DefId, Name),
+    BrNamed(DefId, InternedString),
```

Comment:
Hmm, *this* could affect code, as these names *are* compared from time to time. That said, I don't think we support hygiene on lifetime names, right?

The fact that [this gives an error](https://play.rust-lang.org/?gist=795371edbebb2115a2226ac6fefaca76&version=nightly) suggests I am right.

Maybe @petrochenkov knows.

Reply:
Lifetime names are hygienic in macros 2.0 -- https://play.rust-lang.org/?gist=f3524076b1ad62bacb3979a98ebb72b2&version=nightly

Reply:
Does it matter this late in the pipeline? I thought hygiene would have been satisfied at this point by assigning different `DefId`s.

Reply:
hmm, you might be right.

Reply:
if that is true, then, given that we have the def-id, we could *probably* remove the `InternedString` altogether. The fact that we didn't... well, either an accident, or intentional. I remember @eddyb was playing with this stuff.